client/services/api/api.service.js

```diff
@@ -100,5 +100,15 @@ angular.module('probedock.api').factory('api', function(apiPagination, auth, $ht
     }
   };
 
+  api.slugify = function(str) {
+    return str
+      .replace(/[^a-z0-9\- ]+/gi, '')
+      .replace(/ +/g, '-')
+      .replace(/\-+/g, '-')
+      .replace(/\-+$/, '')
+      .replace(/^\-+/, '')
```

Comment:
As discussed, remove spaces instead of replacing them with hyphens. That will avoid most collisions.

I would replace the whole implementation of this function with:

```
return str.replace(/[^a-z0-9]+/gi, '').toLowerCase()
```

This will keep letters and digits, and nothing else.